config/confighttp/confighttp.go

```diff
@@ -435,3 +440,25 @@ func maxRequestBodySizeInterceptor(next http.Handler, maxRecvSize int64) http.Ha
 		next.ServeHTTP(w, r)
 	})
 }
+
+func memoryLimiterInterceptor(next http.Handler, ext memLimiterFn) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if ext() {
+			http.Error(w, http.StatusText(http.StatusTooManyRequests), http.StatusTooManyRequests)
+			return
+		}
+		next.ServeHTTP(w, r)
+	})
+}
+
+type memLimiterFn = func() bool
+
+func getMemoryLimiterExtension(host component.Host) (memLimiterFn, bool) {
+	extensions := host.GetExtensions()
+	for _, extension := range extensions {
+		if ext, ok := extension.(interface{ MustRefuse() bool }); ok {
```

Comment:
maybe this should have been the previous PR but shouldn't the interface be defined so the interface can be explicitly used?